src/kontrol/kdist/no_stack_checks.md

```diff
@@ -0,0 +1,145 @@
+      [priority(30)]
+
+    rule [super.optimized.mul]:
+      <k> ( #next[ MUL ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => chop ( W0 *Int W1 ) : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.and]:
+      <k> ( #next[ AND ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => W0 &Int W1 : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.or]:
+      <k> ( #next[ EVMOR ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => W0 |Int W1 : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.xor]:
+      <k> ( #next[ XOR ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => W0 xorInt W1 : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.iszero]:
+      <k> ( #next[ ISZERO ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : WS => bool2Word ( W0 ==Int 0 ) : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.lt]:
+      <k> ( #next[ LT ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => bool2Word ( W0 <Int W1 ) : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.gt]:
+      <k> ( #next[ GT ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => bool2Word ( W1 <Int W0 ) : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.eq]:
+      <k> ( #next[ EQ ] => .K ) ... </k>
+      <useGas> false </useGas>
+      <stackChecks> false </stackChecks>
+      <wordStack> W0 : W1 : WS => bool2Word ( W0 ==Int W1 ) : WS </wordStack>
+      <pc> PCOUNT => PCOUNT +Int 1 </pc>
+      [priority(30)]
+
+    rule [super.optimized.next.succ]:
+      <k> ( #next [ OP:OpCode ] => #addr [ OP ] ~> #exec [ OP ] ~> #pc[ OP ] ) ... </k>
+      <stackChecks> false </stackChecks>
+      <wordStack> WS </wordStack>
+      <static> STATIC:Bool </static>
+      requires notBool ( STATIC andBool #changesState(OP, WS) )
+      [priority(31)]
+
+    rule [super.optimized.next.smv]:
+      <k> ( #next [ OP ] => #end EVMC_STATIC_MODE_VIOLATION ) ... </k>
+      <stackChecks> false </stackChecks>
+      <wordStack> WS </wordStack>
+      <static> STATIC:Bool </static>
+      requires STATIC andBool #changesState(OP, WS)
+      [priority(32)]
```

Comment:
Should these be at different priorities relative to each other? I guess the point is that we're more frequently having `STATIC == false`, so setting that rule to try first gives a faster path?

Reply:
I was going off of the fact that it is very unlikely to get a `EVMC_STATIC_MODE_VIOLATION` error from compiled code, so I ordered the priorities accordingly.